src/testdir/test_let.vim

```diff
@@ -93,6 +93,26 @@ func Test_let()
   let [l[0], l[1]] = [10, 20]
   call assert_equal([10, 20, 3], l)
 
+  " Test for using curly brace name in the LHS of an assignment
+  let listvar = [1, 2]
+  let s = 'listvar'
+  let {s} = [3, 4]
+  call assert_equal([3, 4], listvar)
+
+  " Test for using curly brace name as a list and as list index in the LHS of
+  " an assignment
+  let listvar= [1, 2]
```

Comment:
```suggestion
  let listvar = [1, 2]
```

Reply:
Thanks. Updated the PR with this change.